.github/workflows/update-dockerfile-graph.yml

```diff
@@ -0,0 +1,82 @@
+name: Update Dockerfile Dependency Graph
+
+on:
+  push:
+    paths:
+      - 'Dockerfile'
+  pull_request:
+    paths:
+      - 'Dockerfile'
+
+jobs:
+  update-graph:
+    runs-on: ubuntu-latest
+    permissions:
+      contents: write
+      pull-requests: write
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v4
+        with:
+          fetch-depth: 0
+
+      - name: Store old image hash if exists
+        id: old_hash
+        run: |
+          if [ -f "docs/source/assets/contributing/dockerfile-stages-dependency.png" ]; then
+            sha256sum docs/source/assets/contributing/dockerfile-stages-dependency.png > old_hash.txt
+          fi
+
+      - name: Ensure target directory exists
+        run: mkdir -p docs/source/assets/contributing
+
+      - name: Generate Dockerfile graph
+        # The instruction follows https://docs.vllm.ai/en/latest/contributing/dockerfile/dockerfile.html.
+        run: |
+          docker run \
+            --rm \
+            --user "$(id -u):$(id -g)" \
+            --workdir /workspace \
+            --volume "$(pwd)":/workspace \
+            ghcr.io/patrickhoefler/dockerfilegraph:alpine \
+            --output png \
+            --dpi 200 \
+            --max-label-length 50 \
+            --filename Dockerfile \
+            --legend
+          mv Dockerfile.png docs/source/assets/contributing/dockerfile-stages-dependency.png
+
+      - name: Check for changes
+        id: check_changes
+        run: |
+          if [ -f "old_hash.txt" ]; then
+            NEW_HASH="$(sha256sum "docs/source/assets/contributing/dockerfile-stages-dependency.png")"
+            OLD_HASH="$(cat "old_hash.txt")"
+            if [ "$NEW_HASH" != "$OLD_HASH" ]; then
+              echo "changes_detected=true" >> "$GITHUB_OUTPUT"
+              echo "Graph has changed."
+            else
+              echo "changes_detected=false" >> "$GITHUB_OUTPUT"
+              echo "No changes in graph."
+            fi
+          else
+            echo "changes_detected=true" >> "$GITHUB_OUTPUT"
+            echo "First time generating graph."
+          fi
+
+      - name: Create Pull Request
+        if: steps.check_changes.outputs.changes_detected == 'true'
+        uses: peter-evans/create-pull-request@v6
+        with:
+          commit-message: "[Doc] Update Dockerfile dependency graph"
+          title: "[Doc] Update Dockerfile dependency graph"
+          body: |
+            This PR updates the Dockerfile dependency graph following changes to the Dockerfile.
+            
+            - Generated by GitHub Actions workflow
+            - Updates visualization in docs/source/assets/contributing/
+            
+            Signed-off-by: github-actions[bot] <41898282+github-actions[bot]@users.noreply.github.com>
```

Comment:
I carefully designed the title and commit message to resemble the existing ones.